subsys/bluetooth/mesh/net.c

```diff
@@ -783,7 +783,7 @@ int bt_mesh_net_resend(struct bt_mesh_subnet *sub, struct net_buf *buf,
 
 	if (IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY) &&
 	    bt_mesh_proxy_relay(&buf->b, dst)) {
-		net_buf_unref(buf);
+		send_cb_finalize(cb, cb_data);
```

Comment:
The advertiser calls its callbacks only after the packet is actually sent on air, I think we should do the same for proxy, using bt_gatt_notify_cb. There'll be a race between adv and gatt, which complicates things, but we could get rid of the corner cases that caused this bug by generalizing this.

A different PR, probably, though. I can take a look at it if you agree and don't see any big blockers.

Reply:
Yeah, I was thinking of adding a "TODO" note in the code not to forget - it'll require some changes to the pipeline of function calls in proxy.c to get the callback information propagated to the point where it gets passed to the GATT layer using bt_gatt_notify_cb